src/Symfony/Component/Security/Core/Authorization/Voter/AbstractVoter.php

```diff
@@ -83,18 +88,76 @@ public function vote(TokenInterface $token, $object, array $attributes)
     }
 
     /**
-     * Return an array of supported classes. This will be called by supportsClass
+     * Determines if the attribute and class are supported by this voter.
+     *
+     * To determine if the passed class is instance of the supported class, the
+     * isClassInstanceOf() method can be used.
+     *
+     * @param string $attribute An attribute
+     * @param string $class     The fully qualified class name of the passed object
+     *
+     * @return bool True if the attribute and class is supported, false otherwise
+     */
+    protected function supports($attribute, $class)
+    {
+        @trigger_error('The getSupportedClasses and getSupportedAttributes methods are deprecated since version 2.8 and will be removed in version 3.0. Overwrite supports instead.');
```

Comment:
Getting the 2 deprecation messages for the deprecated calls to `getSupported*` are fine IMO, but I think the deprecation warning for this method is not necessary, that was my point.